birdhouse/optional-components/thanos/default.env

```diff
@@ -0,0 +1,50 @@
+
+export THANOS_VERSION=v0.35.1
+export THANOS_DOCKER="thanosio/thanos"
+export THANOS_IMAGE='${THANOS_DOCKER}:${THANOS_VERSION}'
+
+export MINIO_VERSION=RELEASE.2024-05-27T19-17-46Z
+export MINIO_DOCKER=minio/minio
+export MINIO_IMAGE='${MINIO_DOCKER}:${MINIO_VERSION}'
+
+# Minio uses object storage on disk at this location
+export MINIO_DATA_STORE='${BIRDHOUSE_DATA_PERSIST_ROOT}/minio_data/'
+
+# Note that bucket names must only contain lowercase ascii, digits, - and .
+export THANOS_MINIO_BUCKET_NAME=thanos-bucket
+
+# Minio credentials
+export __DEFAULT__MINIO_ROOT_USER=minioadmin
+export __DEFAULT__MINIO_ROOT_PASSWORD=minioadmin
+export MINIO_ROOT_USER="${__DEFAULT__MINIO_ROOT_USER}"
+export MINIO_ROOT_PASSWORD="${__DEFAULT__MINIO_ROOT_PASSWORD}"
+
+# Set a schedule to run the compactor. This should be significantly larger than the PROMETHEUS_LONGTERM_STORE_INTERVAL
+export THANOS_COMPACTOR_WAIT_INTERVAL=24h
+
+# The longterm data retention time can be shortened back to the default since Thanos is now responsible for
+# storing longterm data, not the prometheus-longterm-metrics component.
+export PROMETHEUS_LONGTERM_RETENTION_TIME=15d
+
+# The thanos-sidecar component requires that these two values be equal or else it cannot perform its own compaction
+# https://thanos.io/tip/components/sidecar.md/#sidecar
+export PROMETHEUS_LONGTERM_TSDB_MIN_BLOCK_DURATION=2h
+export PROMETHEUS_LONGTERM_TSDB_MAX_BLOCK_DURATION=2h
+
+VARS="
+  $VARS
+  \$MINIO_ROOT_USER
+  \$MINIO_ROOT_PASSWORD
+"
+
+export DELAYED_EVAL="
+  $DELAYED_EVAL
+  THANOS_IMAGE
+  MINIO_IMAGE
+  MINIO_DATA_STORE
+"
+
+COMPONENT_DEPENDENCIES="
+  ./components/monitoring
```

Comment:
Why does this thanos component depend on the original monitoring component?  It does not add additional configs to the monitoring component.  It uses variables from the monitoring component?

Reply:
It depends on the `prometheus-longterm-metrics` component which depends on the `monitoring` component. But you're right, we don't need to explicitly state this dependency here since it is implied by it's dependence on `prometheus-longterm-metrics`.